### DISCUSSION TITLE
feat: make SubmitAsync<T> method public

### DISCUSSION THREAD

**Discussion by tsanton** [Gremlinq/ExRam.Gremlinq#1560] [Ideas]
**Is your feature request related to a problem? Please describe.**
There are some queries that must be run highly optimised, or that is hard to write/not yet supported by the provider, and it would be great to be able to submit those from raw gremlin. 

**Describe the solution you'd like**
A public method, maybe under the `.AsAdmin()`-api, allowing you to `SubmitAsync<T>`(maybe QueryAsync<T> (with return) and ExecuteAsync(no return). It would be nice if it accepted serialization settings, but all in all it will be fine to just communicate the need for `where T: Vertex` and an expected return format for serialization (maybe `.project('id','label','properties').by(id).by(label).by(map(properties().group().by(key()).by(value())))`)?

**Describe alternatives you've considered**
Continue using the Gremlin.Net.Driver, but it's cumbersome to maintain multiple clients and package versions (as well as conflicts with CosmosDB with the 3.7.* dependency of Gremlin.Net.Driver). 

**Consider sponsoring me**
We'd be happy to purchase some hours of your time to have this available in the near future!

**Comment by danielcweber** [comment]
@tsanton Please have a look at https://docs.gremlinq.net/extensions/, specifically the section about the Gremlinq.Extensions.GroovyScripts package.
